File: 1ºAno/Lista2/Python/Casa/files/ex5.py
total = 0
while True:
    sal = float(input("\nSalário bruto = "))
    total += sal
    if sal == 0:
        print ("Salário líquido total = %.2f" % total)
        break
    h = float(input("Horas trabalhadas = "))
    ant = sal
    if sal >= 800 and sal <= 1600:
        sal -= sal * (0.08 + 0.05)
    elif sal > 1600:
        sal -= sal * (0.15 + 0.07)
    if h > 160:
        sal += (ant / 160) * 0.5 * (h - 160)
    print ("Salário líquido = R$ %.2f" % sal)

File: 1ºAno/Lista2/C/Casa/files/ex5.c
#include <stdio.h>
#include <stdbool.h>

int main() {

    float h, sal, total;

    while (true) {

        printf ("Salário bruto = ");
        scanf ("%f", &sal);
        if (sal == 0) {
            printf ("Salário total = %.2f\n", total);
            break;
        }
        printf ("Horas trabalhadas = ");
        scanf ("%f", &h);
        int ant = sal;
        if (sal >= 800 && sal <= 1600) {
            sal -= 0.08 * sal + 0.05 * sal;
        }
        else if (sal > 1600) {
            sal -= 0.15 * sal + 0.07 * sal;
        }
        if (h > 160) {
            sal += (ant / 160) * 0.5 * (h - 160);
        }
        printf ("Salário líquido = R$ %.2f\n\n", sal);
        total += sal;
    }

    return 0;
}